pandas/core/strings.py

```diff
@@ -1095,24 +1095,48 @@ def str_pad(arr, width, side='left', fillchar=' '):
 
 def str_split(arr, pat=None, n=None):
     """
+    Split strings around given separator/delimiter.
+
     Split each string (a la re.split) in the Series/Index by given
     pattern, propagating NA values. Equivalent to :meth:`str.split`.
 
     Parameters
     ----------
     pat : string, default None
-        String or regular expression to split on. If None, splits on whitespace
+        String or regular expression to split on. If None, split on whitespace.
     n : int, default -1 (all)
-        None, 0 and -1 will be interpreted as return all splits
+        * None, 0 and -1 will be interpreted as return all splits.
+        * Vary output dimensionality if `expand` is True:
+            - If n >= default splits, return all splits.
+            - If n < default splits, makes first n splits only.
     expand : bool, default False
-        * If True, return DataFrame/MultiIndex expanding dimensionality.
+        * If True, return DataFrame/MultiIndex expanding dimensionality.\
+         Appends None for padding.
         * If False, return Series/Index.
 
-    return_type : deprecated, use `expand`
-
     Returns
     -------
     split : Series/Index or DataFrame/MultiIndex of objects
+
+    Examples
+    --------
+    >>> s = pd.Series(["this is good text", "but this is even better"])
+    >>> s.str.split()
```

Comment:
xref my other comment - kind of strange that the method here is actually `str_split` but we are showing how to use `str.split`. While technically the same, from an API perspective I feel like this docstring belongs under the `split` method and we should make this method private internal.

Reply:
same is the case for `str.rsplit`

Reply:
Yep thanks - might be a few other instances in the module as well. I don't want it to hold up what you've done here but could be a good follow up for you to clean up the actual functions

Reply:
Should I work on it in this same pr, or should I make a separate issue for this and work on that?

Reply:
Separate issue